modules/afsocket/afsocket-dest.h

```diff
@@ -69,6 +69,7 @@ struct _AFSocketDestDriver
   LogWriter *(*construct_writer)(AFSocketDestDriver *self);
   gboolean (*setup_addresses)(AFSocketDestDriver *s);
   const gchar *(*get_dest_name)(const AFSocketDestDriver *s);
+  void (*on_connection_restore)(AFSocketDestDriver *s, ReloadStoreItem *ri);
```

Comment:
We chose `ReloadStoreItem`  instead of `LogWriter` because it was not too clear which LogWriter we pass to this function.
This way, it is possible to manipulate with the ReloadStoreItem fields in child classes, which may come in handy in the future (we almost needed something similar to fix this bug).

We can rename it to `restore_connection()`, but the name would be a little false as this function won't restore the connection, it just opens up the the restoration process for extension.

Reply:
We discussed everything IRL, we keep the current behavior and improve the code around ReloadStoreItem a little more.